be/src/exec/pipeline/bucket_process_operator.cpp

```diff
@@ -31,7 +31,7 @@ bool BucketProcessSinkOperator::need_input() const {
 }
 
 bool BucketProcessSinkOperator::is_finished() const {
-    return _ctx->all_input_finishing && _ctx->sink->is_finished();
+    return _ctx->finished || (_ctx->all_input_finishing && _ctx->sink->is_finished());
```

Comment:
why add _ctx->finished ? what if without this condition?

Reply:
This is a quick short-circuit logic Source->JoinProbe If the hash table is empty, our source operator receives set_finished. at this point our BucketSink doesn't need to finish the computation. Without this variable, we will do some unnecessary calculations.